LibDmd/Output/ZeDMD/DMDESP32.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using System.IO.Ports;
+
+namespace DMDESP32
+{
+	public class DMD_ESP32 // Class for locating the COM port of the ESP32 and communicating with it
+	{
+		public string nCOM;
+		public int BaudRate;
+		public byte ByteSize = 8;
+		public bool Opened = false;
+		public byte Parity = 0; // 0-4=no,odd,even,mark,space
+		public string Port;
+		public int ReadTimeout;
+		public byte StopBits = 1; // 0,1,2 = 1, 1.5, 2*/
+		private SerialPort _serialPort;
+		private static readonly int _MAX_SERIAL_WRITE_AT_ONCE = 9400;
+		public static readonly int N_CTRL_CHARS = 6;
+		public static readonly int N_INTERMEDIATE_CTR_CHARS = 4;
+		public static readonly byte[] CtrlCharacters = { (byte)0x5a, (byte)0x65, (byte)0x64, (byte)0x72, (byte)0x75, (byte)0x6d };
+		private bool _Scom_Connect(string port, ref int rx, ref int ry)
+		{
+			try
+			{
+				BaudRate = 921600;
+				Port = port;
+				_serialPort = new SerialPort(port, BaudRate, System.IO.Ports.Parity.None, 8, System.IO.Ports.StopBits.One);
+				_serialPort.ReadTimeout = 100;
+				_serialPort.WriteBufferSize = _MAX_SERIAL_WRITE_AT_ONCE + 100;
+				_serialPort.WriteTimeout = SerialPort.InfiniteTimeout;
+				_serialPort.Open();
+				var result = new byte[Math.Max(N_CTRL_CHARS + 1, N_INTERMEDIATE_CTR_CHARS + 4)];
+				for (int i = 0; i < N_CTRL_CHARS; i++) result[i] = CtrlCharacters[i];
+				result[N_CTRL_CHARS] = 12; // ask for resolution
+				_serialPort.Write(result, 0, N_CTRL_CHARS + 1);
+				System.Threading.Thread.Sleep(100);
+				_serialPort.Read(result, 0, N_INTERMEDIATE_CTR_CHARS + 4);
+				System.Threading.Thread.Sleep(200);

```

Comment:
This is how I would make it more readable:

```suggestion
				_serialPort.Write(CtrlCharacters.Concat(new byte[]{12}).ToArray(), 0, CtrlCharacters.Length + 1);
				System.Threading.Thread.Sleep(100);
				var result = new byte[8];
				_serialPort.Read(result, 0, 8);
				System.Threading.Thread.Sleep(200);

```
Like that you don't have to explicitly copy anything by looping, and you don't have to use `result` for writing the request.